portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+   * Class that manages setting and editing users' nicknames.
+   */
+@WebServlet("/nickname")
+public class NicknameServlet extends HttpServlet {
+
+  /* Get the current nickname, if it exists. Returns an empty string otherwise. 
+     This function is used to set the default value in the form for nickname
+     set up, this way is easier for the user to edit it.*/
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+    response.getWriter().println(nickname);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    if (!userService.isUserLoggedIn()) {
```

Comment:
Optional: Depending on the design of the website it may be unclear why is the form not working. Something to think about whether it will be clear to the user whether what's going on and why they are getting redirected to the index page.

Ideally perhaps, you would not even allow them to fill in the form if they're not authenticated and just return an error otherwise.

Reply:
Actually the form was hidden for the users that are not logged, so I just deleted that.